Firefox Selection Fix.sh

```diff
@@ -66,16 +81,43 @@ function fix_firefox(){
+  # warning [/Applications/Firefox.app/Contents/Resources/browser/omni.ja]:
+  #     48078620 extra bytes at beginning or within zipfile (attempting to process anyway)
+  # error [/Applications/Firefox.app/Contents/Resources/browser/omni.ja]:
+  #     reported length of central directory is -48078620 bytes too long (Atari STZip zipfile?  J.H.Holm ZIPSPLIT 1.1 zipfile?).  Compensating...
   unzip -q "$firefox_dir/browser/omni.ja" -d omni
-  sed -i 's/this\._preventClickSelectsAll = this\.focused;/this._preventClickSelectsAll = true;/' omni/modules/UrlbarInput.jsm
-  sed -i 's/this\._preventClickSelectsAll = this\._textbox\.focused;/this._preventClickSelectsAll = true;/' omni/chrome/browser/content/browser/search/searchbar.js
-  sed -i 's/return shortkey\.split[(]"_"[)]\[1\];/return (shortkey || "_").split("_")[1];/' omni/chrome/devtools/modules/devtools/client/definitions.js # See https://github.com/SebastianSimon/firefox-selection-fix/issues/2
+
+  ### macOS High Sierra sed syntax:
+  # sed [-Ealn] [-e command] [-f command_file] [-i extension] [file ...]
+  # -i extension Edit files in-place, saving backups with the specified extension.
+  #    If a zero-length extension is given, no backup will be saved.
+  sed -i '' 's/this\._preventClickSelectsAll = this\.focused;/this._preventClickSelectsAll = true;/' omni/modules/UrlbarInput.jsm
+  sed -i '' 's/this\._preventClickSelectsAll = this\._textbox\.focused;/this._preventClickSelectsAll = true;/' omni/chrome/browser/content/browser/search/searchbar.js
+  sed -i '' 's/return shortkey\.split[(]"_"[)]\[1\];/return (shortkey || "_").split("_")[1];/' omni/chrome/devtools/modules/devtools/client/definitions.js # See https://github.com/SebastianSimon/firefox-selection-fix/issues/2
+
   cd omni || exit
+
+  ### Official method of creating *.ja files:
+  # https://developer.mozilla.org/en-US/docs/Mozilla/About_omni.ja_(formerly_omni.jar)
   zip -qr9XD omni.ja ./*
+
   cd ..
   mv omni/omni.ja "$firefox_dir/browser/omni.ja" || exit
-  chown --reference=omni.ja~ "$firefox_dir/browser/omni.ja"
-  chmod --reference=omni.ja~ "$firefox_dir/browser/omni.ja"
+
+  ### Copy omni.js backup's permissions to the newly zipped file, in a hopefully platform-independent way
+  # https://unix.stackexchange.com/questions/7730/find-the-owner-of-a-directory-or-file-but-only-return-that-and-nothing-else/7732#7732
+  owner=$(ls -ld omni.ja~ | awk '{print $3}')
+  chown "$owner" "$firefox_dir/browser/omni.ja"
+  # chown --reference=omni.ja~ "$firefox_dir/browser/omni.ja"
+
+  # https://askubuntu.com/questions/152001/how-can-i-get-octal-file-permissions-from-command-line/430697#430697
+  # https://stackoverflow.com/questions/14854265/unix-linux-mac-osx-get-permission-of-file-as-number/14855227#14855227
+  perms=$(ls -l omni.ja~ | awk '{k=0; for(i=0;i<9;++i) k=2*k + (substr($1,i+2,1)!="-"); printf("%0o",k)}')
+  chmod $perms "$firefox_dir/browser/omni.ja"
+  # chmod --reference=omni.ja~ "$firefox_dir/browser/omni.ja"
```

Comment:
Yikes. Isn’t there an easier way, e.g. [Replacement for `chmod --reference` on OS X](https://stackoverflow.com/a/5753809/4642212)?

```suggestion
  chmod $(stat -c '%a' omni.ja~) "$firefox_dir/browser/omni.ja"
```